docs/genai/howto/build-from-source.md

```diff
@@ -38,161 +37,97 @@ These instructions assume you are in the `onnxruntime-genai` folder.
 
+### Build Python API
 
-#### Build ONNX Runtime for CUDA on Windows
+#### Windows CPU build
 
 ```bash
-build.bat --build_shared_lib --skip_tests --parallel --use_cuda --config Release
-copy include\onnxruntime\core\session\onnxruntime_c_api.h ..\onnxruntime-genai\ort\include
-copy include\onnxruntime\core\providers\cuda\*.h ..\onnxruntime-genai\ort\include
-copy build\Windows\Release\Release\*.dll ..\onnxruntime-genai\ort\lib
-copy build\Windows\Release\Release\onnxruntime.lib ..\onnxruntime-genai\ort\lib
+python build.py --config `Release`
 ```
 
-#### Build ONNX Runtime on Linux
+#### Windows DirectML build
 
 ```bash
-./build.sh --build_shared_lib --skip_tests --parallel [--use_cuda] --config Release
-cp include/onnxruntime/core/session/onnxruntime_c_api.h ../onnxruntime-genai/ort/include
-cp build/Linux/Release/libonnxruntime*.so* ../onnxruntime-genai/ort/lib
+python build.py --use_dml --config `Release`
 ```
 
-You may need to provide extra command line options for building with CUDA on Linux. An example full command is as follows.
+#### Linux build
 
 ```bash
-./build.sh --parallel --build_shared_lib --use_cuda --cuda_version 11.8 --cuda_home /usr/local/cuda-11.8 --cudnn_home /usr/lib/x86_64-linux-gnu/ --config Release --build_wheel --skip_tests --cmake_extra_defines CMAKE_CUDA_ARCHITECTURES="80" --cmake_extra_defines CMAKE_CUDA_COMPILER=/usr/local/cuda-11.8/bin/nvcc
+python build.py --config `Release`
 ```
 
-Replace the values given above for different versions and locations of CUDA.
-
-#### Build ONNX Runtime on Mac
+#### Linux CUDA build
 
 ```bash
-./build.sh --build_shared_lib --skip_tests --parallel --config Release
-cp include/onnxruntime/core/session/onnxruntime_c_api.h ../onnxruntime-genai/ort/include
-cp build/MacOS/Release/libonnxruntime*.dylib* ../onnxruntime-genai/ort/lib
+python build.py --use_cuda --config `Release`
 ```
 
-## Build the generate() API
-
-This step assumes that you are in the root of the onnxruntime-genai repo, and you have followed the previos steps to copy the onnxruntime headers and binaries into the folder specified by <ORT_HOME>, which defaults to `onnxruntime-genai/ort`.
+#### Mac build
 
 ```bash
-cd ../onnxruntime-genai
+python build.py --config `Release`
 ```
 
-### Build Python API
-
-#### Build for Windows CPU
-
-```bash
-python build.py
-```
-
-#### Build for Windows DirectML
+### Build Java API
 
 ```bash
-python build.py --use_dml
+python build.py --build_java --config Release
 ```
 
-#### Build on Linux
-
-```bash
-python build.py
-```
+### Build for Android
 
-#### Build on Linux with CUDA
+If building on Windows, install the `ninja` packages.
 
 ```bash
-python build.py --use_cuda
+pip install ninja
 ```
 
-#### Build on Mac
+Run the build script.
 
 ```bash
-python build.py
+python build.py --build_java --android --android_home <path to your Android SDK> --android_ndk_path <path to your NDK installation>` --android_abi --config Release
```

Comment:
I think you need a parameter for android_abi?

Reply:
Fixed